inject.js

```diff
@@ -119,7 +119,7 @@ chrome.runtime.sendMessage({}, function(response) {
 
   function defineVideoController() {
     tc.videoController = function(target, parent) {
-      if (target.dataset['vscid']) {
+      if (target.dataset['vscid'] && document.querySelector(".vsc-controller")) {
```

Comment:
What's the logic here? Or, rather, what edge case is this guarding from?

Reply:
* site removes parent of video element and saves it to anywhere (not DOM)
* therefore mutation.removedNodes not working
* when new video tag created, it uses previous one which has vscid but without vsc-controller
this code part detects this situation and it is not site specific. if there is no controller, create controller normally

Reply:
> 2\. therefore mutation.removedNodes not working

[checkForVideo](https://github.com/igrigorik/videospeed/blob/a687bb1525e30b9fef6c2eb71f2b2d228757334d/inject.js) traverses all the children node, so it should detect wrapped video elements, no?

Also, there can be multiple vsc-controllers on the page, right? Hence my confusion for what this clause is checking: it verifies that there is at least one controller but that doesn't mean that's the "correct" one?

Reply:
I think almost the same but it does not detect deletion for some reason.
Yes there can be multiple videos/vsc-controller. Yes it check if it's at least one on page

Reply:
Can we try and track down why the node is not being detected as removed? The relevant logic is:

https://github.com/igrigorik/videospeed/blob/a687bb1525e30b9fef6c2eb71f2b2d228757334d/inject.js#L340-L354

Reply:
@igrigorik I've looked into this.  Plex is clearing the video element's class list before removing the element.  Because of this, the above code block is being skipped.  Instead of checking if it has that class, could we instead check if it has a vscid dataset?  That isn't being removed from the node, and should continue to work.

Reply:
@vultix yep, I think that's what we landed on here: https://github.com/igrigorik/videospeed/pull/434/files